spring-cloud-security/src/main/java/org/springframework/cloud/security/oauth2/client/ResourceServerTokenRelayAutoConfiguration.java

```diff
@@ -65,20 +69,36 @@
 @ConditionalOnWebApplication
 public class ResourceServerTokenRelayAutoConfiguration {
 
-	@Bean
+	protected static final String TOKEN_RELAY_REQUEST_INTERCEPTOR = "tokenRelayRequestInterceptor";
+
+	@Bean(name = TOKEN_RELAY_REQUEST_INTERCEPTOR)
 	public HandlerInterceptor tokenRelayRequestInterceptor(
 			final OAuth2ClientContext context) {
 		final AccessTokenContextRelay relay = new AccessTokenContextRelay(context);
 		return new HandlerInterceptorAdapter() {
 			@Override
 			public boolean preHandle(HttpServletRequest request,
-					HttpServletResponse response, Object handler) throws Exception {
+									 HttpServletResponse response, Object handler) throws Exception {
 				relay.copyToken();
 				return true;
 			}
 		};
 	}
 
+	@Configuration
+	public static class ResourceServerTokenRelayRegistrationAutoConfiguration extends WebMvcConfigurerAdapter {
+
+		@Autowired
```

Comment:
I see. I guess I'm not really in favour of making the code more complicated just so it can be tested easier. If I were you I'd see if there's a way to just do an integration test.